### DISCUSSION TITLE
ItemsControl  Display  ItemsSource  Item Error

### DISCUSSION THREAD

**Discussion by yangjieshao** [AvaloniaUI/Avalonia#14707] [Q&A]
## Describe the bug
从子线程向 ObservableCollection<T> 添加数据 在 ItemsControl 中会双倍显示
(Adding data from a child thread to ObservableCollection<T>will double the display in itemsControl)

## Screenshots
![image](https://github.com/AvaloniaUI/Avalonia/assets/3275834/31db50b4-fbd4-4bcb-9601-378fec271862)

![image](https://github.com/AvaloniaUI/Avalonia/assets/3275834/81af6709-244e-48ac-87e6-362b500a1329)

- OS: Windows
- Avalonia-Version: 11.0.9

[AvaloniaApplication1.test.zip](https://github.com/AvaloniaUI/Avalonia/files/14367535/AvaloniaApplication1.test.zip)

**Comment by maxkatz6** [comment]
You should update Staffs collection from the UI thread, not from the threadpool/task.run thread.

If would be a better practice, if you do heavy tasks on the separated thread, and return new data to the UI thread before adding it to the collection.

**Reply by kyuranger** [reply]
BTW.When modifying or updating bound data in multi-threads (Task) (not just the ObservableCollection mentioned in this topic, but more string, int, etc.), should I use the UI thread to update, or can it be updated directly in the child thread? I have been very confused about this, although updating directly in the child thread will not report any errors.

**Comment by timunie** [comment]
Try Dispatcher.UiThread.Post to add items.

**Comment by timunie** [comment]
Imo this is not really a bug as we require events to be sent from the UIThread. Will convert it into a Q&A for now. If that wasn't the root cause, we can re-open the issue later.
